### PR TITLE
Add istio deployment

### PR DESCRIPTION
**Pull request by paddatrapper**
Fixes: #585

### CODE REVIEW

File: manifests/istio-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: istio-operator
+  name: istio-operator
+---
+# SYNC WITH manifests/charts/base/files
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
```
Comment (moshloop) [inline manifests/istio-operator.yaml:214]: Can we move the CRD definition out and put it into manifest/crds folder and update the crd phase ?

File: pkg/types/default.go [hunk 2]
```diff
@@ -9,6 +9,9 @@ func DefaultPlatformConfig() PlatformConfig {
 		CertManager: CertManager{
 			Version: "v1.0.3",
 		},
+		IstioOperator: IstioOperator{
```
Comment (moshloop) [inline pkg/types/default.go:12]: I don't think this should be installed by default

Comment (paddatrapper) [reply pkg/types/default.go]: Any suggestions about which test to use?

File: pkg/phases/istiooperator/install.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/flanksource/karina/pkg/platform"
+)
+
+const (
+	Namespace = "istio-operator"
+)
+
+func Install(p *platform.Platform) error {
```
Comment (moshloop) [inline pkg/phases/istiooperator/install.go:11]: I think these need to be added to both cmd.Deploy and cmd.Test

File: manifests/istio-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+      name: istio-operator
+  template:
+    metadata:
+      labels:
+        name: istio-operator
+    spec:
+      serviceAccountName: istio-operator
+      containers:
+        - name: istio-operator
+          image: docker.io/istio/operator:1.8.2
```
Comment (moshloop) [inline manifests/istio-operator.yaml:144]:
```suggestion
          image: docker.io/istio/operator:{{.istioOperator.version | default "na" }}
```

File: pkg/types/default.go (hunk 2 above)
Comment (paddatrapper) [reply pkg/types/default.go]: Use platform

**Comment by paddatrapper** [comment]
Looks like Istio is installed and passes the test in the E2E/test for platform

**Pull request merged**